core/src/main/java/org/jdbi/v3/core/OnDemandExtensions.java

```diff
@@ -51,22 +52,30 @@
                 return extensionType + "@" + Integer.toHexString(System.identityHashCode(proxy));
             }
 
-            try {
-                if (threadExtension.get() != null) {
-                    return method.invoke(threadExtension.get(), args);
-                }
-                return db.withExtension(extensionType, extension ->
-                        JdbiThreadLocals.invokeInContext(threadExtension, extension,
-                                () -> method.invoke(extension, args)));
-            }
-            catch (InvocationTargetException e) {
-                throw e.getTargetException();
+            if (threadExtension.get() != null) {
+                return invoke(method, threadExtension.get(), args);
             }
+            return db.withExtension(extensionType, extension ->
+                    JdbiThreadLocals.invokeInContext(threadExtension, extension,
+                            () -> invoke(method, extension, args)));
         };
 
         return extensionType.cast(
                 Proxy.newProxyInstance(
                         extensionType.getClassLoader(),
                         new Class[]{extensionType}, handler));
     }
+
+    private static Object invoke(Method method, Object target, Object[] args) {
+        try {
+            if (Proxy.isProxyClass(target.getClass())) {
+                return Proxy.getInvocationHandler(target).invoke(target, method, args);
+            }
+            return MethodHandles.lookup().unreflect(method).bindTo(target).invokeWithArguments(args);
```

Comment:
Not sure `MethodHandles.lookup()` buys us anything now with this proxy workaround. Should we just use `method.invoke()` until a corner case presents itself?

Does `MethodHandle` throw the original exception instead of wrapping it?

Reply:
Yes, the only reason I made this change is because it means the exceptions are transparent and it simplifies the catch block, and because we did it in the other place.  I can easily leave the original reflection calls in, at the expense of another nested block for exception unwrapping, if that's preferable.

Reply:
I think I'm going to leave this for now, but I'm happy to do a followup and revert this bit if you think it's too cowboy.